### ISSUE TITLE
Behavior change of FormatDetector.Detect()

### ISSUE THREAD

**Issue by ied206** [B2R2-org/B2R2#27]
**Describe the bug**
B2R2 0.1.x used to return a tuple of `FileFormat` and `ISA` in `FormatDetector.Detect()` method.
However, the method was changed to return only `FileFormat` in the 0.2.x. Is it intentional? If it is, where can I find the substitute for the `ISA` detector method?

I also have looked through the [docs page](https://b2r2.org/APIDoc/api/B2R2.BinFile.FormatDetector.html). It claims that `FormatDetector.Detect()` returns _a tuple of (B2R2.FileFormat and B2R2.ISA)_, but the signature of the method shows only `FileFormat` parameter (`val detect: file:string -> FileFormat`).

**To Reproduce**
In a C# project, copy and paste this code.
Go to the NuGet package manager and downgrade/upgrade B2R2 package, and see which method is allowed per version.
```csharp
// B2R2 0.1.x
(FileFormat format, ISA isa) = FormatDetector.Detect(targetFile);
// B2R2 0.3.0 - where did `ISA isa` go?
FileFormat format = FormatDetector.Detect(targetFile);
```
I believe it is easily reproducible in F#, too.

**Expected behavior**
If the change was intentional, the docs page should be corrected. 
If the ISA detection was not dropped, the docs page should let users know about the alternative API.

**Screenshots**
With `B2R2.FrontEnd` 0.1.1: First method is allowed.
![image](https://user-images.githubusercontent.com/7675657/69342567-2a7f0900-0caf-11ea-9cb0-9646c94845ba.png)

With `B2R2.FrontEnd` 0.3,0: Second method is allowed.
![image](https://user-images.githubusercontent.com/7675657/69342521-13d8b200-0caf-11ea-9bbb-3538b0af285b.png)

[Docs page](https://b2r2.org/APIDoc/api/B2R2.BinFile.FormatDetector.html) has contradicting description and method signature.
![image](https://user-images.githubusercontent.com/7675657/69342175-4fbf4780-0cae-11ea-8f1f-f4ebc6388b00.png)

**Environment (please complete the following information):**
 - OS: Linux Mint 19.2, Windows 10 v1909
 - .NET Core version: 3.0.101
 - B2R2 version 0.3.0
 - Language : C# 

**Additional context**
I originally reviewed applying LowUIR into my project, but had to dismiss the idea because the project was so integrated into the VEX-IR. So for now, I am mainly using B2R2 as a binary `Format` and `ISA` detector. The software capable of detecting binary information is quite a few, alongside (string-parsing-required) libmagic.

P.S. B2R2 프로젝트의 결과물을 모두가 쓸 수 있게 공개해 주셔서 감사합니다.

**EDIT**: Fixed a few typos.

**Comment by sangkilc** [comment]
Thanks for the report.

1. We intentionally changed the API to return only a "FileFormat". So I fixed the doc: b737224c4de4242c06e2bdc7a4788397484b4128

2. You are recommended to use BinHandler to do your job. Inside BinHandler, there is a `FileInfo` field which represents both the `ISA` and the `FileFormat`. To create a BinHandler, you can simply do: `BinHandler.Init (ISA.DefaultISA, filename)` where filename is your binary file path. Note that the first parameter is for a fallback case: it specifies what kind of ISA should be used when the format detection fails.

3. Can I know the reason why you are not using LowUIR?
    >  but had to dismiss the idea because the project was so integrated into the VEX-IR

**Comment by ied206** [comment]
Thanks for updating the docs!

> You are recommended to use BinHandler to do your job

I have read the docs and the code but did not know that using `ISA.DefaultISA` is for a fallback. I will try this method. If you do not mind, I would be pleased if you add some notes on the docs.

> Can I know the reason why you are not using LowUIR?

My project was already tightly coupled with VEX-IR. I tried to implement the LowUIR version but had to change most of my model and approach. It was due to the inherent difference between VEX and LowUIR's design choices.
